app/modules/individuals/models.py

```diff
@@ -302,12 +302,23 @@ def get_names(self):
         return self.names
 
     def get_primary_name(self):
-        if self.names:
-            # Placeholder, first created name to make sure it's always the same one
+        if self.names and self.has_default_name():
+            primary_name = self.get_name_for_context(DEFAULT_NAME_CONTEXT)
+        elif self.names:
             ordered_names = sorted(self.names, key=lambda name: name.created)
-            return ordered_names[0].value
+            primary_name = ordered_names[0].value
         else:
-            return None
+            # should never happen, but seems like the desired behavior if it does in eg a
+            # test where we haven't set names or rendering FE on weirdly-migrated data
+            primary_name = str(self.guid)
+        return primary_name
```

Comment:
This was mostly to get some tests passing without reworking them, was hitting some null issues on existing tests with individuals with no primary names.

Reply:
("this" meaning the final else case where we return the guid)

Reply:
Hm. I could see this being an issue on FE. There are some places where we populate with "Unnamed individual" if getting a name fails.
But, if this is the only way to get tests to pass, I suppose that we could eventually check against the guid in the "Unnamed individual" logic.

Reply:
... is it possible in codex to have an individual without a primary name??
Maybe the tests are the problem?

Reply:
This particular get_primary_name method is only called in the notification builder so I'm guessing FE is categorizing those unnamed individuals by other means.

Reply:
I'd agree that this is a more robust handling of this scenario as it is returning 'something' I'm just wondering if an empty string would be a better somethinng.

Reply:
kinda favoring `None` just based on old-world trauma from there being _both_ empty-string values _and_ nulls all over the place, meaning kind of the same thing.